### PR TITLE
CoM rotation and flip in the GUI

### LINKED ISSUE #31: Visualization and UI for center of mass shift
- [ ] Select rotation, ROI and reference center in detector frame
  - [x] ROI
  - [x] center
  - [ ] handedness
  - [ ] rotation
  - [ ] ring mask option #633 (edit)
- [x] Separate X and Y component image
- [x] Combined image with suitable vector field visualization
- [x] adapt UI for multiple result images per analysis

### PR DESCRIPTION
**Pull request by sk1p**
Fixes #31 

## TODO

- [x] The template for notebook download should also be updated to include the new parameters.
- [x] Fix results data download.
- [x] Add better input method (slider) in addition to allowing to enter values directly
  - [x] Fix for easily entering negative rotation angles in number input
- [x] Add documentation
  - [x]  CoM section in applications section → @uellue will take care of this one
  - [x] Explain parameters (info box or similar?)
- [x] Fix bug: time delta resets after analysis finishes

## Contributor Checklist:

* [x] I have added or updated my entry in [the creators.json file](https://github.com/LiberTEM/LiberTEM/blob/master/packaging/creators.json)
* [x] I have added [a changelog entry](https://github.com/LiberTEM/LiberTEM/tree/master/docs/source/changelog) for my contribution
* [x] I have added/updated documentation for all user-facing changes
* [x] I have added/updated test cases
* [x] I have included the [rebuilt production build of the client](https://libertem.github.io/LiberTEM/contributing.html?#building-the-client) (only if changes were made to the GUI)

## Reviewer Checklist:

* [x] `/azp run libertem.libertem-data` passed

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,7 @@
+root = true
```
Comment (uellue) [inline .editorconfig:1]: Is this file meant to be added?

File: src/libertem/analysis/com.py [hunk 2]
```diff
@@ -295,3 +294,22 @@ def get_parameters(self, parameters):
     @classmethod
     def get_template_helper(cls):
         return ComTemplate
+
```
Comment (uellue) [inline src/libertem/analysis/com.py:297]: OMG it recalculates live when I change the parameters without re-running the UDF? NICE! Can I have a slider for the angle, please? 😁

**Review by uellue — CHANGES_REQUESTED** [review]
Looks good so far! We should include a link or info sign that explains the parameters. We have them documented in https://libertem.github.io/LiberTEM/reference/api.html#libertem.api.Context.create_com_analysis, for example. :-)

With near-instantaneous recalculation after parameter change it would be awesome to have a slider for the angle, or a rotation handle on the widget for position and size!

I noticed that we don't have COM in our Applications section. We could perhaps create a stub for the "phase contrast" application with an API example, link to reference docs and one screenshot for COM, and a link to https://ptychography-4-0.github.io/ptychography/algorithms.html for SSB ptychography?

**Comment by uellue** [comment]
- [ ] The template for notebook download should also be updated to include the new parameters.
- [ ] Fix results data download.

Current result when downloading:

**Comment by uellue** [comment]
We could consider baking a new release soon after this?

**Comment by sk1p** [comment]
> Looks good so far! We should include a link or info sign that explains the parameters. We have them documented in https://libertem.github.io/LiberTEM/reference/api.html#libertem.api.Context.create_com_analysis, for example. :-)

:+1: - I'll add more information about the parameters. I also thought about always showing the parameter form, as it doesn't take up a lot of space anyway - what do you think?

> Current result when downloading: [...]

Thanks, I'll look into that, should be easy to fix.

> We could consider baking a new release soon after this?

Yeah, I thought the same - what do you think about including a fix for #633, too, as I'm touching that code anyways?

> I noticed that we don't have COM in our Applications section. We could perhaps create a stub for the "phase contrast" application with an API example, link to reference docs and one screenshot for COM, and a link to https://ptychography-4-0.github.io/ptychography/algorithms.html for SSB ptychography?

Sure, I'll add a section to the docs.

**Comment by uellue** [comment]
> > Looks good so far! We should include a link or info sign that explains the parameters. We have them documented in https://libertem.github.io/LiberTEM/reference/api.html#libertem.api.Context.create_com_analysis, for example. :-)
> 
> 👍 - I'll add more information about the parameters. I also thought about always showing the parameter form, as it doesn't take up a lot of space anyway - what do you think?

Yes, I was wondering why they were hidden but figured you had your reasons. 😆 
 
> > We could consider baking a new release soon after this?
> 
> Yeah, I thought the same - what do you think about including a fix for #633, too, as I'm touching that code anyways?

Yes, perfect! Would require a bit more change in the CoM API, but should be pretty straightforward, right? :-)

Other points: Thx, great! 👍

**Comment by sk1p** [comment]
@uellue I've pushed the last fixes; if you are happy with this, I can re-build the client and merge - or do you want to push CoM docs also to this PR?

**Comment by uellue** [comment]
> or do you want to push CoM docs also to this PR?

If it is OK it could be a quick solution! :-) I can also rebuild the client if I am at it. :-)

**Comment by sk1p** [comment]
> If it is OK it could be a quick solution! :-) I can also rebuild the client if I am at it. :-)

Sounds good!

**Review by uellue — APPROVED** [review]
Looks good, many thanks! :-)

Should be ready to merge now. :-)

**Pull request merged**